packages/side-nav/src/vaadin-side-nav-item.js

```diff
@@ -7,13 +7,48 @@ import { html, LitElement } from 'lit';
 import { ifDefined } from 'lit/directives/if-defined.js';
 import { ElementMixin } from '@vaadin/component-base/src/element-mixin.js';
 import { PolylitMixin } from '@vaadin/component-base/src/polylit-mixin.js';
+import { SlotController } from '@vaadin/component-base/src/slot-controller.js';
 import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
 import { sideNavItemBaseStyles } from './vaadin-side-nav-base-styles.js';
 
 function isEnabled() {
   return window.Vaadin && window.Vaadin.featureFlags && !!window.Vaadin.featureFlags.sideNavComponent;
 }
 
+/**
+ * A controller to manage the item content children slot.
+ */
+class ChildrenController extends SlotController {
+  constructor(host) {
+    super(host, 'children', null, {
+      observe: true,
+      multiple: true,
+    });
+  }
+
+  /**
+   * @protected
+   * @override
+   */
+  initAddedNode(node) {
+    const { host } = this;
+
+    host._childElements.add(node);
```

Comment:
I'd avoid modifying the host's properties from controllers where possible as it introduces implicit dependencies. Would it work if we checked the nodes array directly from the component?

```js
updated() {
  ...
  if (this._childrenController.nodes.length > 0) {
    this.toggleAttribute('has-children', this._childElements.size > 0);
  }
}
```


```